### PR TITLE
Support ``networkx.Graph``s as input to ``Lattice``

### LINKED ISSUE #465: Initialize ``Lattice``s with ``networkx.Graph``s
### What should we add?

It would be nice to allow the initialization on ``Lattice``s not only with retworkx graphs but also with networks graphs. Networkx, to date, still has a lot more features and pre-defined graphs that users have access to, and is probably _the_ standard graph library in Python.

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Closes #465.

### Details and comments

Split 2/2 of PR #467.

### CODE REVIEW

**Review by mrossinek — COMMENTED** [review]
This already looks good. I will review this properly once #471 is merged, since this is blocked by that PR.

**Review by mrossinek — COMMENTED** [review]
I do not have a real opinion on this but I guess more options are good for users. Once the CI passes I am fine with merging this :+1: 

But @pbark should review this as he has more knowledge about the lattice module.

File: qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py
Comment (Cryoris) [inline qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:120]: Probably this type hint should be a string since `nx.Graph` is not defined if users don't have it installed..

File: qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py
Comment (mrossinek) [inline qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:27]: I guess you can refactor this in-line with #535

File: qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py
Comment (mrossinek) [reply qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:120]: Maybe take a look at how #535 deals with that. In the `PyQuanteDriver` something similar to this may occur. Maybe @manoelmarques can also tell you directly :+1:

Comment (manoelmarques) [reply qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:120]: Yes, strings like it is already done for the `Matplotlib` type hints in `LatticeDrawStyle`

File: qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py
Comment (manoelmarques) [reply qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:27]: To be really lazy, ideally the import  should be moved inside the method it is used and the method marked with:
```
@_optionals.HAS_NETWORKX.require_in_call
```
If that is not possible then the non lazy option but still complying with the new Terra API is to change the top with:
```
if _optionals.HAS_NETWORKX:
    # pylint: disable=unused-import
    import networkx as nx
```

And in the method where you would check the HAS_NX, just add:
```
@_optionals.HAS_NETWORKX.require_in_call
```
or if it is a `init` call, use it in the class definition:
```
@_optionals.HAS_NETWORKX.require_in_instance
```

Comment (Cryoris) [reply qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:27]: Ok, then I'll wait for #535 to be merged 👍🏻 Thanks Manoel!

Comment (manoelmarques) [reply qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py:27]: In this case the `HAS_NETWORKX` is from Terra so you don't need the Nature changes but they have been merged.

**Review by mrossinek — DISMISSED** [review]

**Review by mrossinek — APPROVED** [review]

**Review by mrossinek — DISMISSED** [review]

**Review by mrossinek — APPROVED** [review]

**Review by ikkoham — APPROVED** [review]
LGTM. Thanks

**Pull request merged**
